Bottom: 039938181ed773367fb1a1d7f84f745b5cb868b5
Top:    6a3ee6aaf48cfecc309f714a6b79cf0938a1bec8
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-07-14 20:57:46 +0800

vhost_net: honor exported skb array when doing busy polling

Signed-off-by: Jason Wang <jasowang@redhat.com>


---

diff --git a/drivers/vhost/net.c b/drivers/vhost/net.c
index e3d7ea1..b51a7df 100644
--- a/drivers/vhost/net.c
+++ b/drivers/vhost/net.c
@@ -601,10 +601,13 @@ static int peek_head_len(struct vhost_net_virtqueue *rvq, struct sock *sk)
 	return len;
 }
 
-static int sk_has_rx_data(struct sock *sk)
+static int sk_has_rx_data(struct vhost_net_virtqueue *vq, struct sock *sk)
 {
 	struct socket *sock = sk->sk_socket;
 
+	if (vq->rx_array)
+		return vhost_net_buf_peek(vq);
+
 	if (sock->ops->peek_len)
 		return sock->ops->peek_len(sock);
